src/top/state.rs

```diff
@@ -1,19 +1,28 @@
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, HashMap};
 
 use tokio::sync::mpsc;
+use vector_core::internal_event::DEFAULT_OUTPUT;
 
 use crate::config::ComponentKey;
 
 type IdentifiedMetric = (ComponentKey, i64);
 
+#[derive(Debug)]
+pub struct SentEventsMetric {
+    pub key: ComponentKey,
+    pub total: i64,
+    pub outputs: HashMap<String, i64>,
+}
+
 #[derive(Debug)]
 pub enum EventType {
     ReceivedEventsTotals(Vec<IdentifiedMetric>),
     /// Interval in ms + identified metric
     ReceivedEventsThroughputs(i64, Vec<IdentifiedMetric>),
-    SentEventsTotals(Vec<IdentifiedMetric>),
-    /// Interval in ms + identified metric
-    SentEventsThroughputs(i64, Vec<IdentifiedMetric>),
+    // Identified overall metric + metric broken out by output
```

Comment:
This comment seems off.

Reply:
Ah, the wording might be confusing. I phrased it as such because the new `SentEventsMetric` struct can contain both a `total` overall metric and an `outputs` map that maps `output name -> output-specific metric` for a given component. Would `Identified overall metric + output-specific metrics` be clearer?

Reply:
👍🏻 Yep, that works for me. It was the "metric + metric" that threw me.